File: Java/RecursiveCall/BJAlgo_6603.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class BJAlgo_6603 {
    static StringBuilder sb = new StringBuilder();

    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        while (true) {
            sb.setLength(0);
            int[] inputs = Arrays.stream(br.readLine().split(" ")).mapToInt(Integer::parseInt).toArray();

            if (inputs.length == 1 && inputs[0] == 0) {
                break;
            }

            int[] nums = new int[inputs.length - 1];
            System.arraycopy(inputs, 1, nums, 0, nums.length);

            lotto(nums, 0, nums.length - 5, new ArrayList());

            System.out.println(sb);
        }
    }

    private static void lotto(int[] nums, int start, int end,  ArrayList<Integer> selected) {
        if (selected.size() == 6) {
            for (int i = 0; i < 5; i++) {
                sb.append(selected.get(i)).append(" ");
            }
            sb.append(selected.get(5)).append("\n");
            return;
        }

        for (int i = start; i < end; i++) {
            if (i < nums.length) {
                selected.add(nums[i]);
                lotto(nums, i + 1, i + (nums.length - 5) + 1, selected);
                selected.remove(selected.size() - 1);
            }
        }
    }
}

File: Python/RecursiveCall/BJAlgo_6603.py
from sys import stdin
import copy

result = list() # 하나의 케이스의 모든 경우의 수를 담을 리스트


def lotto(nums, selected):
    if len(selected) == 6:
        result.append(copy.deepcopy(selected)) # 6개의 수가 모두 선택된 경우 해당 경우를 result 리스트에 추가
        return

    for i in range(len(nums)):
        selected.append(nums[i]) # 숫자를 하나 뽑아 선택된 리스트에 추가하고
        lotto(nums[(i + 1):], selected) # 그 숫자 이후 리스트를 다시 반복
        selected.pop() # 해당 자리 수의 숫자를 빼기


while True:
    result.clear()
    inputs = list(map(int, stdin.readline().split()))

    # 입력이 0인 경우 종료
    if len(inputs) == 1 and inputs[0] == 0:
        break

    k = inputs[0]
    s = inputs[1:]

    lotto(s, [])

    for r in result:
        print(" ".join(str(n) for n in r))

    print()
